build/ci/compliance.yml

```diff
@@ -0,0 +1,50 @@
+# Name: DotNet-Project-System-Compliance
+# URL: https://devdiv.visualstudio.com/DevDiv/_build?definitionId=15013
+#
+# Responsible for running compliance checks.
+
+#
+# NOTE: triggers for this build are defined in the Web UI instead of here in the YAML file so they
+#       apply to all branches.
+
+resources:
+- repo: self
+  clean: true
+queue:
+  name: VSEngSS-MicroBuild2019-1ES
+  demands: Cmd
+  timeoutInMinutes: 90
+variables:
+  BuildConfiguration: Release
+  TeamName: DotNet-Project-System
+  BuildPlatform: any cpu
+  DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true
+
+steps:
+- script: $(Build.SourcesDirectory)\build.cmd /build /no-test /ci /sign /diagnostic /no-deploy /no-integration /no-ibc /no-clearnugetcache /configuration $(BuildConfiguration)
+  displayName: Build ProjectSystem.sln
+
+- task: PoliCheck@2
+  inputs:
+    targetType: 'F'                             # Scan a particular file or folder (recursively)
+    targetArgument: '$(Build.SourcesDirectory)' # Path of file/folder to scan
+    result: 'PoliCheck.xml'                     # Name of the output file
+    optionsFC: '1'                              # Enable scanning of comments
+    optionsSEV: '1|2|3'                         # Scan for severity 1, 2, and 3 issues
+
+- task: CredScan@3
+  inputs:
+    outputFormat: 'pre' # Ouput in PREFast format so TSAUpload can consume it
+
+- task: APIScan@2
+  inputs:
+    softwareFolder: '$(Build.SourcesDirectory)\artifacts'
+    softwareName: 'Dotnet-Project-System'
+    softwareVersionNum: '17.0'
+    softwareBuildNum: '$(Build.BuildId)'
+    symbolsFolder: 'SRV*http://symweb'
+
+- task: TSAUpload@2
```

Comment:
What is TSAUpload? I'm having difficulty finding docs on it.

Reply:
It picks up the output files from the previous stages and uses them to auto-generate bugs.